TTS/tts/datasets/dataset.py

```diff
@@ -776,3 +804,155 @@ def print_logs(self, level: int = 0) -> None:
         print("\n")
         print(f"{indent}> F0Dataset ")
         print(f"{indent}| > Number of instances : {len(self.samples)}")
+
+
+class EnergyDataset:
```

Comment:
It might actually be easier to compute Energy in ```model.format_batch_on_device``` using ```torch_transforms``` or the functions in the ```vits.py``` since it is quite fast to compute on GPU with torch primitives and we avoid caching that way. 

Hope it makes sense.

Reply:
@erogol yes it makes sense , i was doing this to preserve a same format between pitch and energy, if i write it in ```vits.py``` then it'll be limited to ```vits.py``` only so i have put it in to dataset.

Reply:
@erogol yes it makes sense, i was doing this to preserve a same format between pitch and energy, if i write it in ```vits.py``` then it'll be limited to ```vits.py```, so i have put it in dataset.